apps/rush-lib/src/api/Rush.ts

```diff
@@ -53,7 +54,9 @@ export class Rush {
   public static launch(launcherVersion: string, arg: ILaunchOptions): void {
     const options: ILaunchOptions = Rush._normalizeLaunchOptions(arg);
 
-    Rush._printStartupBanner(options.isManaged);
+    if (!Utilities.shouldRestrictConsoleOutput()) {
```

Comment:
This is a somewhat awkward design.  Seems like the problem you are solving is that Rush normally prints output like the startup banner, but for certain CLI commands we want to suppress all output except for a specific data structure.

Seems like a better long-term design would be for `ts-command-line`'s API to be refactored into several steps:

1. **Parsing:** Silently analyze the `argv` array but do not print any output.  Errors would simply be recorded in an API variable.
2. Here is a chance for certain commands like `rush tab-complete` to bypass the normal `ts-command-line` framework and interpret the parser results however they like.
3. **Execute:** Then the object-oriented scaffold runs normally, invoking `CommandLineAction.onExecute()` for the subclass.

Right now Step 1+3 happen together inside `CommandLineParser.executeWithoutErrorHandling()`

Reply:
BTW if we replace all the `console.log()` calls with @iclanton's `Terminal` system, then we could eliminate all these `Utilities.shouldRestrictConsoleOutput()` checks by categorically suppressing/redirecting the Terminal so it's no longer connected to STDOUT.  

Probably out of scope for this PR, though.

Reply:
@octogonz Shall we do this as a separate PR?

Reply:
Yes